### PR TITLE
feat(Table): Added ability to hide columns and new Editable collection

### PR DESCRIPTION
**Pull request by MagicAardvark**
##### **Description**
Added new options for hiding  cell values or complete columns, used on edit vs view.

Added new collection which has a switchable datasource allowing us to 'cache' a data source before entering edit mode.

##### **Reviewers**
* @jgodi
* @more

### CODE REVIEW

**Comment by jgodi** [comment]
@bvkimball ^^

File: src/elements/table/Table.ts [hunk 1]
```diff
@@ -270,7 +270,7 @@ export class NovoTableElement implements DoCheck {
     @Output() onRowSelect: EventEmitter<any> = new EventEmitter();
     @Output() onTableChange: EventEmitter<any> = new EventEmitter();
 
-    _dataProvider: PagedCollection<any>;
+    _dataProvider: any;
```
Comment (bvkimball) [inline src/elements/table/Table.ts]: this is going backwards...  this needs to be an Interface, Collection maybe

File: src/elements/table/Table.ts [hunk 2]
```diff
@@ -882,6 +884,7 @@ export class NovoTableElement implements DoCheck {
                 row._editing[column.name] = false;
             });
         });
+        this._dataProvider.leaveEditMode();
```
Comment (bvkimball) [inline src/elements/table/Table.ts]: Not all dataProviders have this functionality

File: src/elements/table/Table.ts [hunk 3]
```diff
@@ -975,7 +979,7 @@ export class NovoTableElement implements DoCheck {
      * @memberOf NovoTableElement
      */
     cancelEditing(): void {
-        this.dataProvider.refresh();
+        this.dataProvider.cancelEditMode();
```
Comment (bvkimball) [inline src/elements/table/Table.ts]: Again not all DataProviders have this.

File: src/elements/table/Table.ts [hunk 4]
```diff
@@ -1017,4 +1021,26 @@ export class NovoTableElement implements DoCheck {
...
     }
+
+    /**
+     * @name hideValue
+     * @description hide a column value in edit or view mode
+     * @param {column meta} column
+     * @returns {boolean}
+     * @memberOf NovoTableElement
+     */
+    hideValue(column: any): boolean {
```
Comment (bvkimball) [inline src/elements/table/Table.ts]: isValueHidden

File: src/elements/table/Table.ts [hunk 5]
```diff
@@ -1017,4 +1021,26 @@ export class NovoTableElement implements DoCheck {
...
+    }
+
+    /**
+     * @name hideColumn
+     * @description hide a column in edit or view mode
+     * @param {column meta} column
+     * @returns {boolean}
+     * @memberOf NovoTableElement
+     */
+    hideColumn(column: any): boolean {
```
Comment (bvkimball) [inline src/elements/table/Table.ts]: isColumnHidden

File: src/elements/table/Table.ts [hunk 6]
```diff
@@ -184,9 +184,9 @@ export class NovoTableFooterElement {
...
                                 </td>
-                                <td *ngFor="let column of columns" [attr.data-automation-id]="column.id || column.name" [class.novo-form-row]="editable">
-                                    <novo-table-cell *ngIf="!row._editing[column.name]" [hasEditor]="editable" [column]="column" [row]="row" [form]="tableForm.controls.rows.controls[i]"></novo-table-cell>
-                                    <novo-control *ngIf="row._editing[column.name]" condensed="true" [form]="tableForm.controls.rows.controls[i]" [control]="row.controls[column.name]" [hidden]="column.hiddenOnEdit"></novo-control>
+                                <td *ngFor="let column of columns" [attr.data-automation-id]="column.id || column.name" [class.novo-form-row]="editable" [hidden]="hideColumn(column)">
+                                    <novo-table-cell *ngIf="!row._editing[column.name] || !column.editor" [hasEditor]="editable" [column]="column" [row]="row" [form]="tableForm.controls.rows.controls[i]" [hidden]="hideValue(column)"></novo-table-cell>
```
Comment (bvkimball) [inline src/elements/table/Table.ts]: Seem like a lot of duplicate logic going on here...

File: src/elements/table/Table.ts (hunk 1 above)
Comment (MagicAardvark) [inline src/elements/table/Table.ts]: problem is it can be any one of serval collections

**Pull request merged**